res/res_pjsip_logger.c

```diff
@@ -665,15 +668,27 @@ static void check_debug(void)
 	RAII_VAR(char *, debug, ast_sip_get_debug(), ast_free);
 
 	if (ast_false(debug)) {
-		pjsip_disable_logger(-1);
+		/* If the logger was enabled via the CLI instead of
+		 * through the config file, then we shouldn't
+		 * disable it on a reload.
+		 * Only disable logging if the logger was previously
+		 * enabled via the config file. */
+		if (logger_enabled_via_config) {
+			pjsip_disable_logger(-1);
```

Comment:
This would not handle the case if you do CLI based logging and config based logging simultaneously and change the config based logging. It would turn off logging.